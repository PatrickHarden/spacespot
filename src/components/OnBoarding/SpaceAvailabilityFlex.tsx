import React, { useState } from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { useIntl } from 'react-intl'
import styled from 'styled-components'
import actions from 'services/onboarding/actions'
import selectors from 'services/onboarding/selectors'
import {
  SpaceData,
  AvailabilityFlex,
  AvailabilityFlexPriceInterval,
} from 'services/onboarding/types'
import {
  getRegionCurrencyDesc,
  getRegionCurrencyCode,
} from 'services/global/region'
import { invalidDate, invalidNumber } from 'services/global/util'

import Input from 'components/common/Input'
import DatePicker from 'components/common/DatePicker'
import Heading3 from 'components/common/Heading3'
import Heading4 from 'components/common/Heading4'
import ServicedOffices from './ServicedOffices'
import moment from 'moment'

const Header4 = styled(Heading4)`
  margin-top: 0;
`
const Container = styled.div`
  max-width: 516px;
  box-sizing: border-box;
  display: flex;
  flex-wrap: wrap;
  border-radius: 2px;
  background-color: #f4f4f4;
  padding: 20px;
  margin-bottom: 50px;

  @media (max-width: 767px) {
    padding: 15px;
  }
`

const ContainerInputs = styled.div`
  padding: 10px 15px;
  border: 1px dashed #dddddd;
  background-color: #ffffff;
  flex-grow: 1;
  margin-right: 10px;
  :last-child {
    margin-right: 0;
    margin-left: 10px;
  }
  width: calc(50% - 42px);
  max-width: 330px;

  @media (max-width: 480px) {
    width: 100%;
    margin: 0 0 20px;
    :last-child {
      margin: 0;
    }
  }
`

const getDefaultFields = (value?: AvailabilityFlex) => {
  const fields: { [key: string]: string } = {}
  if (value) {
    fields.desks = value.desks ? value.desks.toString() : ''
    fields.minLease = value.minLease ? value.minLease.toString() : ''
    fields.price = value.price ? value.price.toString() : ''
    fields.availableFrom = value.availableFrom.toString()
  } else {
    const defaultDate = new Date()
    defaultDate.setDate(new Date().getDate() + 1)
    fields.desks = ''
    fields.minLease = ''
    fields.price = ''
    fields.availableFrom = defaultDate.toString()
  }
  return fields
}

const SpaceAvailabilityFlex = (props: {
  space: SpaceData
  setError: (id: string | null) => void
}) => {
  const dispatch = useDispatch()
  const { formatMessage } = useIntl()
  const t = (s: string) => formatMessage({ id: s })
  const state = useSelector(selectors.onboardingState)
  const country = state ? state.country : ''

  const { space, setError } = props

  const hotFields = getDefaultFields(space.hotDesks)
  const [hotData, setHotData] = useState(hotFields)
  const [showHotErrors, setShowHotErrors] = useState(false)
  const fixedFields = getDefaultFields(space.fixedDesks)
  const [fixedData, setFixedData] = useState(fixedFields)
  const [showFixedErrors, setShowFixedErrors] = useState(false)

  const valForm = (fields: { [key: string]: string }) => {
    const errors: { [key: string]: string } = {}
    if (!fields.desks) fields.desks = '0'
    if (invalidNumber(fields.desks, 0)) errors.desks = t('VAL_INVALID')
    if (Number(fields.desks) > 0) {
      if (invalidNumber(fields.minLease, 1)) errors.minLease = t('VAL_INVALID')
      if (invalidNumber(fields.price, 0)) errors.price = t('VAL_INVALID')
      if (invalidDate(fields.availableFrom))
        errors.availableFrom = t('VAL_INVALID')
    }
    return errors
  }
  const fixedErrors = showFixedErrors ? valForm(fixedData) : {}
  const hotErrors = showHotErrors ? valForm(hotData) : {}

  const mapToAvailability = (data: {
    [key: string]: string
  }): AvailabilityFlex => {
    const desks = Number(data.desks)
    return {
      desks,
      minLease: desks > 0 ? Number(data.minLease) : 0,
      price: desks > 0 ? Number(data.price) : 0,
      availableFrom: new Date(data.availableFrom),
      currencyCode: getRegionCurrencyCode(country),
      frequency: AvailabilityFlexPriceInterval.Monthly,
    }
  }

  const hotVal = (newData: { [key: string]: string }) => {
    const errors = valForm(newData)
    if (Object.keys(errors).length > 0) {
      setShowHotErrors(true)
      setError('hot-desks-' + Object.keys(errors)[0])
      return
    }
    setError(null)
    const hotDesks = mapToAvailability(newData)
    dispatch(actions.setSpaceHotDesks(space.id, hotDesks))
  }

  const hotChange = (field: string) => (
    e: React.ChangeEvent<HTMLInputElement>,
  ) => {
    const val = e.target.value
    const newData = { ...hotData, [field]: val }
    setHotData(newData)
    hotVal(newData)
  }
  const hotChangeDate = (field: string) => (d?: string | null) => {
    if (d) {
      d = moment(d, 'DD/MM/YYYY').format('MM/DD/YYYY') as string
      const newData = { ...hotData, [field]: d }
      setHotData(newData)
      hotVal(newData)
    }
  }

  const fixedVal = (newData: { [key: string]: string }) => {
    const errors = valForm(newData)
    if (Object.keys(errors).length > 0) {
      setShowFixedErrors(true)
      setError('fixed-desks-' + Object.keys(errors)[0])
      return
    }
    setError(null)
    const fixedDesks = mapToAvailability(newData)
    dispatch(actions.setSpaceFixedDesks(space.id, fixedDesks))
  }

  const fixedChange = (field: string) => (
    e: React.ChangeEvent<HTMLInputElement>,
  ) => {
    const val = e.target.value
    const newData = { ...fixedData, [field]: val }
    setFixedData(newData)
    fixedVal(newData)
  }
  const fixedChangeDate = (field: string) => (d?: string | null) => {
    if (d) {
      d = moment(d, 'DD/MM/YYYY').format('MM/DD/YYYY') as string
      const newData = { ...fixedData, [field]: d }
      setFixedData(newData)
      fixedVal(newData)
    }
  }

  const zeroToEmpty = (val: string) => {
    return val === '0' ? '' : val
  }

  return (
    <>
      <Heading3>{t('ONBOARDING_SPACE_AVAILABILITY_FLEX_TITLE')}</Heading3>
      <Container>
        <ContainerInputs>
          <Header4>{t('ONBOARDING_SPACE_FLEX_HOT_DESKS')}</Header4>
          <Input
            id="hot-desks-desks"
            data-testid="hot-desks-desks"
            type="number"
            label={t('ONBOARDING_SPACE_FLEX_DESKS')}
            onChange={hotChange('desks')}
            value={zeroToEmpty(hotData.desks)}
            error={hotErrors.desks}
          />
          <DatePicker
            format="dd.MM.yyyy"
            id="hot-desks-availableFrom"
            data-testid="hot-desks-from"
            label={t('ONBOARDING_SPACE_FLEX_AVAILABILITY')}
            value={new Date(hotData.availableFrom)}
            onChange={hotChangeDate('availableFrom')}
            disablePast={true}
            error={hotErrors.availableFrom}
          />
          <Input
            id="hot-desks-minLease"
            data-testid="hot-desks-min-lease"
            type="number"
            label={t('ONBOARDING_SPACE_FLEX_MINLEASE')}
            onChange={hotChange('minLease')}
            value={hotData.minLease}
            error={hotErrors.minLease}
          />
          <Input
            id="hot-desks-price"
            data-testid="hot-desks-price"
            type="number"
            label={`${t(
              'ONBOARDING_SPACE_FLEX_PRICE',
            )} (${getRegionCurrencyDesc(country)})`}
            onChange={hotChange('price')}
            placeholder={t('ONBOARDING_PLACEHOLDER_FLEX_PRICE')}
            value={hotData.price}
            error={hotErrors.price}
          />
        </ContainerInputs>
        <ContainerInputs>
          <Header4>{t('ONBOARDING_SPACE_FLEX_FIXED_DESKS')}</Header4>
          <Input
            id="fixed-desks-desks"
            data-testid="fixed-desks-desks"
            type="number"
            label={t('ONBOARDING_SPACE_FLEX_DESKS')}
            onChange={fixedChange('desks')}
            value={zeroToEmpty(fixedData.desks)}
            error={fixedErrors.desks}
          />
          <DatePicker
            format="dd.MM.yyyy"
            id="fixed-desks-availableFrom"
            data-testid="fixed-desks-from"
            label={t('ONBOARDING_SPACE_FLEX_AVAILABILITY')}
            value={new Date(fixedData.availableFrom)}
            onChange={fixedChangeDate('availableFrom')}
            disablePast={true}
            error={fixedErrors.availableFrom}
          />
          <Input
            id="fixed-desks-minLease"
            data-testid="fixed-desks-min-lease"
            type="number"
            label={t('ONBOARDING_SPACE_FLEX_MINLEASE')}
            onChange={fixedChange('minLease')}
            value={fixedData.minLease}
            error={fixedErrors.minLease}
          />
          <Input
            id="fixed-desks-price"
            data-testid="fixed-desks-price"
            type="number"
            label={`${t(
              'ONBOARDING_SPACE_FLEX_PRICE',
            )} (${getRegionCurrencyDesc(country)})`}
            onChange={fixedChange('price')}
            placeholder={t('ONBOARDING_PLACEHOLDER_FLEX_PRICE')}
            value={fixedData.price}
            error={fixedErrors.price}
          />
        </ContainerInputs>
      </Container>
      <ServicedOffices space={space} />
    </>
  )
}
export default SpaceAvailabilityFlex
