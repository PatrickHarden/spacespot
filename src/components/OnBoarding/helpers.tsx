import {
  OnBoardingState,
  SpaceData,
  SpaceStatus,
  AvailabilityFlexPriceInterval,
} from 'services/onboarding/types'

const date = new Date('2020/01/01')

export const createSpace = (type: string): SpaceData => ({
  id: '0',
  type,
  spaceName: '',
  spaceDescription: '',
  spaceHighlights: '',
  spaceSize: undefined,
  spaceFloor: undefined,
  spaceSizeCommon: undefined,
  spaceMatterPort: '',
  spaceFloored: '',
  use: '',
  spaceRent: undefined,
  spaceServices: undefined,
  spaceServicesNotNegotiable: false,
  availabilityFixed: date,
  months: undefined,
  hotDesks: {
    desks: 0,
    availableFrom: date,
    minLease: 0,
    price: 0,
    currencyCode: 'NOK',
    frequency: AvailabilityFlexPriceInterval.Monthly,
  },
  fixedDesks: {
    desks: 0,
    availableFrom: date,
    minLease: 0,
    price: 0,
    currencyCode: 'NOK',
    frequency: AvailabilityFlexPriceInterval.Monthly,
  },
  servicedOffices: [],
  images: [],
  floorPlan: [],
  fitout: [{ description: 'FitOut1', name: 'FitOut1', amount: 1000 }],
  status: SpaceStatus.ReadyForPublish,
})

export const createOnBoardingState = (): OnBoardingState => ({
  isNew: true,
  buildingId: undefined,
  images: [
    {
      key: 0,
      name: '40VilliersStreet.External_Photo_1_small.png',
      type: 'image/png',
      uri:
        'data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mP8/5+hHgAHggJ/PchI7wAAAABJRU5ErkJggg==',
    },
  ],
  docs: [],
  name: '',
  description: '',
  address: '',
  city: '',
  postCode: '',
  country: '',
  location: '',
  latLng: undefined,
  district: '',
  subdistrict: {
    id: 0,
    name: '',
  },
  amenities: { hasGym: { checked: true, desc: 'Gym' } },
  spaces: { '0': createSpace('FLEX') },
  spaceCount: 1,
})

export const valInteger = (
  current: number | undefined,
  value: number,
  min: number,
  max?: number,
): number => {
  const val = Number(Number(value).toFixed())
  return val < min || (max && val > max) ? current || 0 : val
}
