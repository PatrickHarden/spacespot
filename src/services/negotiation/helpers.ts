import { get } from 'lodash'
import {
  NegotiationTerms,
  NegotiationTerm,
  NegotiationTermFields,
  NegotiationTermFieldsIds,
} from './types'

export const getNegotiationTermArray = (
  terms: NegotiationTerms,
  ids: NegotiationTermFieldsIds,
) => {
  const values: Array<NegotiationTerm> = Object.keys(terms).map(key => {
    return {
      termId: ids[key],
      value: get(terms, key),
    }
  })
  return values
}

const decapitalize = (title: string) => title[0].toLowerCase() + title.slice(1)

export const getFields = (
  fields: NegotiationTermFields,
  negTerms: Array<NegotiationTerm>,
): NegotiationTerms => {
  const initialTerms = {} as NegotiationTerms
  if (!negTerms || !fields) return initialTerms
  return (negTerms.reduce(
    (acc, term) => ({
      ...acc,
      [decapitalize(fields[term.termId])]: term.value,
    }),
    initialTerms,
  ) as unknown) as NegotiationTerms
}
