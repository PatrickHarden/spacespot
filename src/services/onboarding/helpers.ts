import { get, cloneDeep } from 'lodash'
import { Space } from 'services/space/types'
import spaceSelectors from 'services/space/selectors'
import {
  AmenityOptions,
  OnBoardingState,
  AvailabilityFlexPriceInterval,
  AvailabilityFlex,
} from 'services/onboarding/types'

export const aspects2Amenities = (
  amenities: AmenityOptions,
  aspects: Array<string>,
): AmenityOptions => {
  const newAmenities = cloneDeep(amenities) || {}
  if (!aspects) {
    return newAmenities
  }
  aspects.forEach(key => {
    if (newAmenities[key]) {
      newAmenities[key].checked = true
    } else {
      newAmenities[key] = {
        checked: true,
        desc: key,
      }
    }
  })
  return newAmenities
}

export const loadOnboardingSpace = (
  defaultAmenities: AmenityOptions | undefined,
  space: Space,
  isBuilding: boolean,
): OnBoardingState => {
  const building = spaceSelectors.translateBuilding(space, isBuilding)
  return {
    buildingId: building.buildingId,
    isNew: false,
    images: building.images || [],
    docs: [],
    name: building.name,
    description: building.description,
    address: building.address,
    city: building.city,
    postCode: building.postCode,
    country: building.country,
    location: building.location || '',
    latLng: building.latLng,
    district: building.district,
    subdistrict: building.subdistrict,
    amenities: aspects2Amenities(
      defaultAmenities || {},
      get(space, 'Common.Aspects', []),
    ),
    spaces: isBuilding ? {} : spaceSelectors.translateSpace(space),
    spaceCount: 0,
    uploadedImages: building.uploadedImages,
    uploadedDocs: building.uploadedDocs,
    isPublishing: false,
    defaultAmenities: defaultAmenities,
  }
}

const defaultDate = new Date()
defaultDate.setDate(new Date().getDate() + 1)

export const getDefaultAvailability = (): AvailabilityFlex => ({
  desks: 0,
  availableFrom: defaultDate,
  minLease: 0,
  price: 0,
  currencyCode: 'NOK',
  frequency: AvailabilityFlexPriceInterval.Monthly,
})
