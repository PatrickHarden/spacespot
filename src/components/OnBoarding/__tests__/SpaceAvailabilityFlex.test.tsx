import React from 'react'
import SpaceAvailabilityFlex from '../SpaceAvailabilityFlex'
import { render, fireEvent } from '@testing-library/react'
import { createSpace } from '../helpers'
import MockDate from 'mockdate'

jest.mock('react-intl', () => ({
  ...jest.requireActual('react-intl'),
  useIntl: () => ({ formatMessage: (s: any) => s.id }),
}))

const mockDispatch = jest.fn()
jest.mock('react-redux', () => ({
  useDispatch: () => {
    return mockDispatch
  },
  useSelector: () => () => '',
}))

const emptyDesks = {
  id: '0',
  value: {
    availableFrom: jasmine.anything(),
    currencyCode: jasmine.anything(),
    desks: 0,
    minLease: 0,
    price: 0,
    frequency: 'Monthly',
  },
}

it('Renders without crashing ', async () => {
  MockDate.set(1434319925275)
  const spaceMock = createSpace('')
  const wrapper = render(
    <SpaceAvailabilityFlex space={spaceMock} setError={jest.fn()} />,
  )
  expect(wrapper.asFragment()).toMatchSnapshot()
})

it('Hot desks edit', async () => {
  const dispatch = mockDispatch
  MockDate.reset()
  const spaceMock = createSpace('')
  const wrapper = render(
    <SpaceAvailabilityFlex space={spaceMock} setError={jest.fn()} />,
  )
  const desks = wrapper.getByTestId('hot-desks-desks')
  const minLease = wrapper.getByTestId('hot-desks-min-lease')
  const price = wrapper.getByTestId('hot-desks-price')
  fireEvent.change(minLease, { target: { value: 12 } })
  expect(dispatch).toHaveBeenLastCalledWith({
    payload: emptyDesks,
    type: 'ONBOARDING/SET_SPACE_HOT_DESKS',
  })
  fireEvent.change(price, { target: { value: 2000 } })
  expect(dispatch).toHaveBeenLastCalledWith({
    payload: emptyDesks,
    type: 'ONBOARDING/SET_SPACE_HOT_DESKS',
  })
  fireEvent.change(desks, { target: { value: 10 } })
  expect(dispatch).toHaveBeenLastCalledWith({
    payload: {
      id: '0',
      value: {
        availableFrom: jasmine.anything(),
        currencyCode: jasmine.anything(),
        desks: 10,
        minLease: 12,
        price: 2000,
        frequency: 'Monthly',
      },
    },
    type: 'ONBOARDING/SET_SPACE_HOT_DESKS',
  })
})

it('Fixed desks edit', async () => {
  const dispatch = mockDispatch
  MockDate.reset()
  const spaceMock = createSpace('')
  const wrapper = render(
    <SpaceAvailabilityFlex space={spaceMock} setError={jest.fn()} />,
  )
  const desks = wrapper.getByTestId('fixed-desks-desks')
  const minLease = wrapper.getByTestId('fixed-desks-min-lease')
  const price = wrapper.getByTestId('fixed-desks-price')
  fireEvent.change(minLease, { target: { value: 12 } })
  expect(dispatch).toHaveBeenLastCalledWith({
    payload: emptyDesks,
    type: 'ONBOARDING/SET_SPACE_FIXED_DESKS',
  })
  fireEvent.change(price, { target: { value: 2000 } })
  expect(dispatch).toHaveBeenLastCalledWith({
    payload: emptyDesks,
    type: 'ONBOARDING/SET_SPACE_FIXED_DESKS',
  })
  fireEvent.change(desks, { target: { value: 10 } })
  expect(dispatch).toHaveBeenLastCalledWith({
    payload: {
      id: '0',
      value: {
        availableFrom: jasmine.anything(),
        currencyCode: jasmine.anything(),
        desks: 10,
        minLease: 12,
        price: 2000,
        frequency: 'Monthly',
      },
    },
    type: 'ONBOARDING/SET_SPACE_FIXED_DESKS',
  })
})
