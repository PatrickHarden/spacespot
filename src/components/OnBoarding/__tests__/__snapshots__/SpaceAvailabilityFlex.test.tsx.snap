// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing  1`] = `
<DocumentFragment>
  <h3
    class="sc-EHOje gPIdwP"
  >
    ONBOARDING_SPACE_AVAILABILITY_FLEX_TITLE
  </h3>
  <div
    class="sc-eNQAEJ bkVpIq"
  >
    <div
      class="sc-hMqMXs kCoLBW"
    >
      <h4
        class="sc-bZQynM sc-jKJlTe gLHKEV"
      >
        ONBOARDING_SPACE_FLEX_HOT_DESKS
      </h4>
      <div
        class="sc-bdVaJa dlaMlh"
      >
        <label
          for="hot-desks-desks"
        >
          ONBOARDING_SPACE_FLEX_DESKS
        </label>
        <input
          class="sc-bwzfXH jHsaxX"
          data-testid="hot-desks-desks"
          id="hot-desks-desks"
          type="number"
          value=""
        />
        <div
          class="sc-htpNat gctktF"
        />
      </div>
      <div
        class="sc-bxivhb fuvWwV sc-ifAKCX gRKhFU"
      >
        <label>
          ONBOARDING_SPACE_FLEX_AVAILABILITY
        </label>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              id="hot-desks-availableFrom"
              type="text"
              value="2020-01-01"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <img
                    alt="open calendar"
                    src="calendar.svg"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-60 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-61"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="sc-bdVaJa dlaMlh"
      >
        <label
          for="hot-desks-minLease"
        >
          ONBOARDING_SPACE_FLEX_MINLEASE
        </label>
        <input
          class="sc-bwzfXH jHsaxX"
          data-testid="hot-desks-min-lease"
          id="hot-desks-minLease"
          type="number"
          value=""
        />
        <div
          class="sc-htpNat gctktF"
        />
      </div>
      <div
        class="sc-bdVaJa dlaMlh"
      >
        <label
          for="hot-desks-price"
        >
          ONBOARDING_SPACE_FLEX_PRICE (Kr)
        </label>
        <input
          class="sc-bwzfXH jHsaxX"
          data-testid="hot-desks-price"
          id="hot-desks-price"
          placeholder="ONBOARDING_PLACEHOLDER_FLEX_PRICE"
          type="number"
          value=""
        />
        <div
          class="sc-htpNat gctktF"
        />
      </div>
    </div>
    <div
      class="sc-hMqMXs kCoLBW"
    >
      <h4
        class="sc-bZQynM sc-jKJlTe gLHKEV"
      >
        ONBOARDING_SPACE_FLEX_FIXED_DESKS
      </h4>
      <div
        class="sc-bdVaJa dlaMlh"
      >
        <label
          for="fixed-desks-desks"
        >
          ONBOARDING_SPACE_FLEX_DESKS
        </label>
        <input
          class="sc-bwzfXH jHsaxX"
          data-testid="fixed-desks-desks"
          id="fixed-desks-desks"
          type="number"
          value=""
        />
        <div
          class="sc-htpNat gctktF"
        />
      </div>
      <div
        class="sc-bxivhb fuvWwV sc-ifAKCX gRKhFU"
      >
        <label>
          ONBOARDING_SPACE_FLEX_AVAILABILITY
        </label>
        <div
          class="MuiFormControl-root MuiTextField-root"
        >
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
              id="fixed-desks-availableFrom"
              type="text"
              value="2020-01-01"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
            >
              <button
                class="MuiButtonBase-root MuiIconButton-root"
                tabindex="0"
                type="button"
              >
                <span
                  class="MuiIconButton-label"
                >
                  <img
                    alt="open calendar"
                    src="calendar.svg"
                  />
                </span>
                <span
                  class="MuiTouchRipple-root"
                />
              </button>
            </div>
            <fieldset
              aria-hidden="true"
              class="PrivateNotchedOutline-root-60 MuiOutlinedInput-notchedOutline"
              style="padding-left: 8px;"
            >
              <legend
                class="PrivateNotchedOutline-legend-61"
                style="width: 0.01px;"
              >
                <span>
                  ​
                </span>
              </legend>
            </fieldset>
          </div>
        </div>
      </div>
      <div
        class="sc-bdVaJa dlaMlh"
      >
        <label
          for="fixed-desks-minLease"
        >
          ONBOARDING_SPACE_FLEX_MINLEASE
        </label>
        <input
          class="sc-bwzfXH jHsaxX"
          data-testid="fixed-desks-min-lease"
          id="fixed-desks-minLease"
          type="number"
          value=""
        />
        <div
          class="sc-htpNat gctktF"
        />
      </div>
      <div
        class="sc-bdVaJa dlaMlh"
      >
        <label
          for="fixed-desks-price"
        >
          ONBOARDING_SPACE_FLEX_PRICE (Kr)
        </label>
        <input
          class="sc-bwzfXH jHsaxX"
          data-testid="fixed-desks-price"
          id="fixed-desks-price"
          placeholder="ONBOARDING_PLACEHOLDER_FLEX_PRICE"
          type="number"
          value=""
        />
        <div
          class="sc-htpNat gctktF"
        />
      </div>
    </div>
  </div>
  <h3
    class="sc-EHOje gPIdwP"
  >
    ONBOARDING_SPACE_FLEX_SERVICED_OFFICES
  </h3>
  <div
    class="sc-kGXeez iKQQkx"
  >
    <div
      class="sc-kpOJdX dPnLnX"
    >
      <div
        class="sc-bdVaJa dlaMlh sc-jzJRlG sc-cSHVUG fLZezk"
      >
        <label
          for="serviced-offices-desks"
        >
          ONBOARDING_SPACE_SO_PEOPLE
        </label>
        <input
          class="sc-bwzfXH jHsaxX"
          id="serviced-offices-desks"
          min="1"
          type="number"
          value=""
        />
        <div
          class="sc-htpNat gctktF"
        />
      </div>
      <div
        class="sc-bdVaJa dlaMlh sc-jzJRlG sc-cSHVUG fLZezk"
      >
        <label
          for="serviced-offices-floor"
        >
          ONBOARDING_SPACE_SO_FLOOR
        </label>
        <input
          class="sc-bwzfXH jHsaxX"
          id="serviced-offices-floor"
          type="number"
          value=""
        />
        <div
          class="sc-htpNat gctktF"
        />
      </div>
      <div
        class="sc-dxgOiQ bhiYeY"
      >
        <div
          class="sc-bxivhb fuvWwV sc-ifAKCX gRKhFU"
        >
          <label>
            ONBOARDING_SPACE_FLEX_AVAILABILITY
          </label>
          <div
            class="MuiFormControl-root MuiTextField-root"
          >
            <div
              class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd"
            >
              <input
                aria-invalid="false"
                class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd"
                id="serviced-offices-from"
                type="text"
                value="2015-06-15"
              />
              <div
                class="MuiInputAdornment-root MuiInputAdornment-positionEnd"
              >
                <button
                  class="MuiButtonBase-root MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <span
                    class="MuiIconButton-label"
                  >
                    <img
                      alt="open calendar"
                      src="calendar.svg"
                    />
                  </span>
                  <span
                    class="MuiTouchRipple-root"
                  />
                </button>
              </div>
              <fieldset
                aria-hidden="true"
                class="PrivateNotchedOutline-root-60 MuiOutlinedInput-notchedOutline"
                style="padding-left: 8px;"
              >
                <legend
                  class="PrivateNotchedOutline-legend-61"
                  style="width: 0.01px;"
                >
                  <span>
                    ​
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="sc-bdVaJa dlaMlh sc-jzJRlG eKZRLG"
        >
          <label
            for="serviced-offices-min-lease"
          >
            ONBOARDING_SPACE_FLEX_MINLEASE
          </label>
          <input
            class="sc-bwzfXH jHsaxX"
            id="serviced-offices-min-lease"
            min="1"
            type="number"
            value=""
          />
          <div
            class="sc-htpNat gctktF"
          />
        </div>
      </div>
      <div
        class="sc-bdVaJa dlaMlh sc-jzJRlG sc-cSHVUG fLZezk"
      >
        <label
          for="serviced-offices-price"
        >
          ONBOARDING_SPACE_SO_PRICE (Kr)
        </label>
        <input
          class="sc-bwzfXH jHsaxX"
          id="serviced-offices-price"
          min="0"
          placeholder="ONBOARDING_SPACE_SO_PRICE_PLACEHOLDER"
          type="number"
          value=""
        />
        <div
          class="sc-htpNat gctktF"
        />
      </div>
      <div
        class="sc-ckVGcZ bMnoqN"
      >
        <button
          class="sc-gzVnrw ilVVTZ"
        >
          ONBOARDING_SPACE_SO_SAVE
        </button>
        <button
          class="sc-gzVnrw sc-kAzzGY dqiIbv"
        >
          ONBOARDING_SPACE_SO_CANCEL
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
